packages/shared/src/components/modals/SteppedModal.tsx

```diff
@@ -3,37 +3,86 @@ import React, {
   KeyboardEvent,
   MouseEvent,
   ReactElement,
+  useContext,
   useState,
 } from 'react';
+import AnalyticsContext from '../../contexts/AnalyticsContext';
+import useAuthForms from '../../hooks/useAuthForms';
+import { AuthEventNames } from '../../lib/auth';
+import { logout } from '../../lib/user';
+import AuthOptions, { AuthDisplay } from '../auth/AuthOptions';
+import { DiscardAuthModal } from '../auth/common';
 import { Button } from '../buttons/Button';
 import TabContainer, { Tab } from '../tabs/TabContainer';
+import { SimpleTooltip } from '../tooltips/SimpleTooltip';
 import { ModalProps, StyledModal } from './StyledModal';
 
+type ValidateFunction = () => boolean | Promise<boolean>;
+
 interface SteppedModalProps extends ModalProps {
+  trigger: string;
+  skippable?: boolean;
+  invalidMessage?: string;
   isLastStepLogin?: boolean;
+  isFirstStepIntroduction?: boolean;
+  onValidateNext?: ValidateFunction[];
   onStepChange?: (stepBefore: number, stepNow: number) => void | Promise<void>;
+  onInvalid?: (step: number) => void | Promise<void>;
   onFinish?: () => void | Promise<void>;
+  onAuthSuccess?: (isLogin?: boolean) => void | Promise<void>;
 }
 
 function SteppedModal({
+  trigger,
   children,
+  skippable = true,
+  invalidMessage,
   isLastStepLogin,
+  isFirstStepIntroduction,
+  onValidateNext = [],
+  onInvalid,
   onStepChange,
   onFinish,
   onRequestClose,
+  onAuthSuccess,
   ...props
 }: SteppedModalProps): ReactElement {
```

Comment:
`SteppedModal` sounds like a generic component but looking inside, it seems like it's not.
Should we change the name of the component?

Reply:
It is actually a generic component, you will have to put multiple children, and each will be considered a step, this has just abstracted the whole moving forward and backward steps.

Reply:
Have now refactored the component to be more loose so the consuming end will have better flexibility, which in turn will have less abstraction which I think was a good part.